worker/src/sender/mod.rs

```diff
@@ -0,0 +1,39 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use proto::common::{
+    instance::{self, InstanceId},
+    node::Metrics,
+};
+use reqwest::Client;
+
+/// Sends a http request to `ctl::http` component with body
+/// containing `instance::Status` information used in `ctl::discovery`
+#[derive(Clone, Debug)]
+pub struct Sender {
+    pub client: Client,
+    pub http_path: SocketAddr,
+}
+
+impl Sender {
+    /// Creates a `Sender` instance wrapped by an `Arc`.
+    #[must_use]
+    pub fn new(path: SocketAddr) -> Arc<Self> {
+        Arc::new(Sender {
+            client: Client::new(),
+            http_path: path,
+        })
+    }
+    /// The `ctl::http` URL should be known by the worker since the start.
+    /// So it should be passed via `args` to the main function.  
```

Comment:
This comment is not related to the `send_status` method

Reply:
I've placed it over `http_path`, any other suggestion?